R/cor_df.R

```diff
@@ -123,7 +124,7 @@ rplot.cor_df <- function(rdf,
   row_order <- rdf$rowname
 
   # Convert data to relevant format for plotting
-  pd <- stretch(rdf, na.rm = TRUE)
+  pd <- stretch(rdf, na.rm = TRUE, .order = .order)
```

Comment:
`stretch` does not need to be modified necessarily. (I'm adding another comment to the proposed changes in the body of the function). 

The call to `stretch` itself seems to be fine as it preserves the order in both x and y columns as is. It is only necessary to use factors internal to `rplot` to get `ggplot` to override its default alphabetical ordering. 

The following lines between the calls to `stretch` and `ggplot` within `rplot` will get the plot to reflect the order of the variables in the correlation matrix. 

```
  if(.order != "alphabet"){
    if(.order == "default"){
      pd$x <- factor(pd$x, levels = row_order)
      pd$y <- factor(pd$y, levels = rev(row_order))
    }
  }
```
I've structured the above branching logic to accommodate possible future non-alphabetical choices of ordering.

Someone using `ggplot` manually with a data.frame output from stretch would need to replicate the factor conversion. Which is standard. My preference is for type stability/consistency of `stretch`. 

I think changing the factors internally in `rplot` is appropriate as it is a plot meant specifically to represent a correlation matrix as the matrix has been received.

Reply:
Ah, I got what you were saying now. 👍